### PR TITLE
cli: Reject higher majors from this main line update, tbd a new command flag

### PR DESCRIPTION
**Pull request by nfuden**
Avoid allowing glooctl to install v2 variants without an explicit call for it.
      This may be reverted later once we have a better understanding of upgrade path.

Testing by looking the test.sh file and modifying it to prove out the filtering works

### CODE REVIEW

**Review by sam-heilbron — COMMENTED** [review]
lgtm

File: projects/gloo/cli/pkg/cmd/upgrade/cmd.go [hunk 1]
```diff
@@ -138,6 +138,11 @@ func getReleaseWithAsset(ctx context.Context, httpClient *http.Client, tag strin
 			if err != nil {
 				continue
 			}
+
+			if v.Major != 1{
```
Comment (sam-heilbron) [inline projects/gloo/cli/pkg/cmd/upgrade/cmd.go:142]: nit: Will this fail codegen?

Comment (sam-heilbron) [reply projects/gloo/cli/pkg/cmd/upgrade/cmd.go:142]: nit: A comment here would be really helpful I think, so people know when we can remove this

File: projects/gloo/cli/test.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
```
Comment (sam-heilbron) [inline projects/gloo/cli/test.sh:1]: Is this run in CI?

**Review by sam-heilbron — APPROVED** [review]

File: projects/gloo/cli/test.sh (hunk 2 above)
Comment (nfuden) [reply projects/gloo/cli/test.sh:1]: This should be used to validate our changes

**Review by sam-heilbron — APPROVED** [review]

File: projects/gloo/cli/pkg/cmd/upgrade/cmd.go (hunk 1 above)
Comment (nfuden) [reply projects/gloo/cli/pkg/cmd/upgrade/cmd.go]: Unsure at this time. We should work with others to understand if we can ever remove it from v1 branches

**Review by EItanya — APPROVED** [review]

**Pull request merged**